operator/v1/types.go

```diff
@@ -50,29 +50,45 @@ type ClusterManagerSpec struct {
 	// DeployOption contains the options of deploying a cluster-manager
 	// Default mode is used if DeployOption is not set.
 	// +optional
+	// +kubebuilder:default={mode: Default}
 	DeployOption DeployOption `json:"deployOption,omitempty"`
 }
 
-type InstallMode string
-
-const (
-	InstallModeDefault  InstallMode = "Default"
-	InstallModeDetached InstallMode = "Detached"
-)
-
+// DeployOption describes the deploy options for cluster-manager or klusterlet
 type DeployOption struct {
 	// Mode can be Default or Detached.
-	// In Default mode, the Hub is installed as a whole and all parts of Hub are deployed in the same cluster.
-	// In Detached mode, only crd and configurations are installed on one cluster(defined as hub-cluster). Controllers run in another cluster (defined as management-cluster) and connect to the hub with the kubeconfig in secret of "external-hub-kubeconfig"(a kubeconfig of hub-cluster with cluster-admin permission).
+	// For cluster-manager:
+	//   - In Default mode, the Hub is installed as a whole and all parts of Hub are deployed in the same cluster.
+	//   - In Detached mode, only crd and configurations are installed on one cluster(defined as hub-cluster). Controllers run in another cluster (defined as management-cluster) and connect to the hub with the kubeconfig in secret of "external-hub-kubeconfig"(a kubeconfig of hub-cluster with cluster-admin permission).
+	// For klusterlet:
+	//   - In Default mode, all klusterlet related resources are deployed on the managed cluster.
+	//   - In Detached mode, only crd and configurations are installed on the spoke/managed cluster. Controllers run in another cluster (defined as management-cluster) and connect to the mangaged cluster with the kubeconfig in secret of "external-managed-kubeconfig"(a kubeconfig of managed-cluster with cluster-admin permission).
```

Comment:
@zhujian7 how do we specify the configuration of management-cluster to controllers?

Reply:
In fact for registration-agent and work-agent, the configuration here specifically refers to the RBAC configuration. Currently, we don't need a configuration resource(cm) for the controllers(registration-agent and work-agent), all configs are reflected by the flags.